app/controllers/Leagues.java

```diff
@@ -13,214 +11,213 @@
+		case 1:
+			league.setNumTeams(tempForm.get().numTeams);
+			league.setNumGames(tempForm.get().numGames);
+			league.setRegStep(2);
+			break;
+		case 2:
+			league.setDateList(tempForm.get().dateList);
+			league.setRegStep(3);
+			break;
+		case 3:
+			List<String> teamList = tempForm.get().teams;
+
+			for(int i = 0; i < teamList.size(); i++){
+				if(!league.getTeams().contains(Team.getTeam(teamList.get(i)))){
+					league.addTeam(Team.getTeam(teamList.get(i)));
+				}
+			}
+			league.eraseSchedule();
+			generateSchedule(league);
+			league.setRegStep(4);
+			break;
+		case 4:
+			league.setCourt(Court.getCourt(tempForm.get().court));
+			league.setDescription(tempForm.get().description);
+			league.setRegStep(5);
+			break;
+		default:
+			break;
+		}
+
+
+		LeagueDB.addLeague(league);
+		LeagueForm leagueForm = new LeagueForm(league);
+		Form<LeagueForm> form = Form.form(LeagueForm.class).fill(leagueForm);
+		return ok(LeagueList.render("League Team", league, form, league.getTeams(), Secured.isLoggedIn(ctx())));
+	}
+
+	public static Result addTeam(long id){
+		League league = LeagueDB.getLeague(id);
+		Form<LeagueForm> tempForm = Form.form(LeagueForm.class).bindFromRequest();
+		league.addTeam(Team.getTeam(tempForm.get().teams.get(0)));
+		league.eraseSchedule();
+		generateSchedule(league);
+		LeagueDB.addLeague(league);
+		LeagueForm leagueForm = new LeagueForm(league);
+		Form<LeagueForm> form = Form.form(LeagueForm.class).fill(leagueForm);
+		return ok(LeagueList.render("League Team", league, form, league.getTeams(), Secured.isLoggedIn(ctx())));
+	}
+
+	public static Result changeSchedule(){
+		League league = LeagueDB.getLeague(1);
+
+		Form<LeagueForm> leagueForm = Form.form(LeagueForm.class).bindFromRequest();
+
+		if (leagueForm.hasErrors()) {
+			return badRequest(LeagueSchedule.render("Create Schedule", leagueForm, Secured.isLoggedIn(ctx())));
+		}
+		else {
+			LeagueForm schedule = leagueForm.get();
+			//League.addSchedule(schedule);
+			List<Team> listTeam = league.getTeams();
+			return ok(LeagueList.render("League Team", league, leagueForm, listTeam, Secured.isLoggedIn(ctx())));
+		}
+	}
+
+	@Security.Authenticated(Secured.class)
+	public static Result addLeague(){
+		Form<LeagueForm> leagueForm = Form.form(LeagueForm.class).bindFromRequest();
+		List<Team> listTeam = LeagueDB.getLeague(1).getTeams();
+
+		if (leagueForm.hasErrors()) {
+			return badRequest(LeagueList.render("League Team", LeagueDB.getLeague(1), leagueForm, listTeam, Secured.isLoggedIn(ctx())));
+		}
+		else {
+			LeagueForm form = leagueForm.get();
+			LeagueDB.addLeague(form);
+			if(LeagueDB.getLeague(leagueForm.get().id) == null){
+				listTeam = LeagueDB.getTeamsInLeague(1);
+			}
+			else{
+				listTeam = LeagueDB.getTeamsInLeague(leagueForm.get().id);
+			}
+
+			//return ok(LeagueList.render("League Team", LeagueDB.getLeague(leagueForm.get().id), leagueForm, listTeam, Secured.isLoggedIn(ctx())));
+			return redirect("/leagues/view/" + form.id);
```

Comment:
should probably use the ```return redirect(routes.Leagues.league(form.id))``` (I think that's how it is) here, just in case we decide to change the path.